.github/workflows/sync.yml

```diff
@@ -0,0 +1,194 @@
+name: Sync
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+env:
+  BOT_USERNAME: "exercism-bot"
+  GIT_USERNAME: "Exercism Bot"
+  GIT_EMAIL: "66069679+exercism-bot@users.noreply.github.com"
+
+jobs:
+  fetch-tracks:
+    name: Fetch & fork tracks
+    runs-on: ubuntu-20.04
+
+    outputs:
+      tracks: ${{ steps.fetch-tracks.outputs.result }}
+
+    steps:
+      - name: Fetch list of tracks
+        id: fetch-tracks
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          script: |
+            const trackRepos = await github.paginate(github.search.repos, {
+              q: 'org:exercism+topic:exercism-track'
+            })
+
+            //return trackRepos.flatMap(({ full_name }) => [full_name])
+
+            // TODO: Before merge, remove this and uncomment the line above
+            return ['exercism/julia']
+
+      - name: Determine tracks needing to be forked
+        id: required-forks
+        uses: actions/github-script@47f7cf65b5ced0830a325f705cad64f2f58dddf7
+        with:
+          github-token: ${{ secrets.BOT_PERSONAL_ACCESS_TOKEN }}
+          script: |
+            const botRepos = await github.paginate(github.repos.listForAuthenticatedUser, {
+              affiliation: 'owner'
+            })
+
+            // Replacing the bot username with exercism is slightly hacky, but necessary
+            // for the comparison below. Unfortunately, GitHub's API response does not seem
+            // to contain any reference to the upstream repo of the fork.
+            const botForks = botRepos.flatMap(({ full_name, fork }) => fork ? [full_name.replace(process.env.BOT_USERNAME, 'exercism')] : [])
```

Comment:
The graphql API _does_ allow for referencing the upstream repo, which they refer to as the "parent". E.g., the following graphql will output all the forked repos with their forked name and upstream name:

```graphql
{
  user(login: "ErikSchierboom") {
    repositories(first: 100, isFork: true) {
      nodes {
        nameWithOwner
        parent {
          nameWithOwner
        }
      }
    }
  }
}
```

Which outputs:

```json
{
  "data": {
    "user": {
      "repositories": {
        "nodes": [
          {
            "nameWithOwner": "ErikSchierboom/fsharp",
            "parent": {
              "nameWithOwner": "exercism/fsharp"
            }
          },
          {
            "nameWithOwner": "ErikSchierboom/problem-specifications",
            "parent": {
              "nameWithOwner": "exercism/problem-specifications"
            }
          },
          ...
        ]
      }
    }
  }
}
```

You can try this yourself at https://docs.github.com/en/graphql/overview/explorer

According to the docs, `actions/github-script` allows you to run graphql queries: https://github.com/actions/toolkit/tree/master/packages/github

The main disadvantage is that you'd have to do manual pagination, so it might be not that convenient after all.

Reply:
That would be useful, I'll take a look.

Reply:
Let's leave this open as a future TODO for now, I'm not sure if the added complexity is worth it, and I don't know graphql at all.

Reply:
Sure